autogen/code_utils.py

```diff
@@ -228,17 +228,27 @@ def get_powershell_command():
             if result.returncode == 0:
                 return "pwsh"
 
-        except (FileNotFoundError, NotADirectoryError):
-            if WIN32:
+        except (FileNotFoundError, NotADirectoryError, PermissionError):
+            if PermissionError:
+                return "The application has no permission to run powershell"
```

Comment:
You should log the error and return `None`. The contract of this function is if a value is returned that is used as the powershell command. So in this case the command used will be "The application has no permission to run powershell"